extensions/wikia/FacebookClient/scripts/FacebookClient.XFBML.js

```diff
@@ -2,19 +2,34 @@
 
 /**
  * Render FB XML tags emebedded into wikitext
- * This file is loaded via ResourceLoader only when FB XML tags are present on the page
  */
 $(function () {
 	'use strict';
 
 	// load api on page load and on demand
 	mw.hook('wikipage.content').add(function ($content) {
-		$.loadFacebookAPI()
-			.done(function () {
-				$('.sso-login').removeClass('hidden');
 
-				// scan the new content for any fb tags
-				FB.XFBML.parse($content[0]);
-			});
+		if ((xfbmlTagsOnPage())) {
+			if (facebookSDKNotLoaded()) {
+				// XFBML tags rendered automatically when SDK loads
+				$.loadFacebookSDK();
+			} else {
+				renderXFBMLTags();
+			}
+		}
+
+		function xfbmlTagsOnPage() {
+			var numOfXFBMLTags = $content.find('[data-type="xfbml-tag"]').length;
```

Comment:
I've always written this kind of thing like `return $content.find('[data-type="xfbml-tag"]').length`, although that will return a number instead of a boolean, so that might not be as clear. But type casting like `return !!$content.find('[data-type="xfbml-tag"]').length` is also acceptable, as is your currently solution, which is arguably the most clear, but also seems unnecessarily verbose to someone that's used to working with jQuery.